### PR TITLE
RGBDS syntax updates

### PR DESCRIPTION
**Pull request by vulcandth**
Implementing the new RGBDS Macro Syntax. This was done using regex. It `make compare` `OK`. Although, Please review thoroughly to make sure I didn't change something I shouldn't have.

Please also reference: #686

- [x] RGBDS PR for Pokecrystal
- [x] RGBDS PR for Pokegold: https://github.com/pret/pokegold/pull/78
- [x] RGBDS PR for Pokered: https://github.com/pret/pokered/pull/358
- [x] RGBDS PR for Pokeyellow: https://github.com/pret/pokeyellow/pull/86
- [x] RGBDS PR for Poketcg: https://github.com/pret/poketcg/pull/116 
- [x] RGBDS PR for Pokepinball: https://github.com/pret/pokepinball/pull/35 
- [x] RGBDS Commit for Poketcg2: https://github.com/pret/poketcg2/commit/764c416acd42e14df90c60d78e1b582ce0d6f92e

### CODE REVIEW

**Comment by Rangi42** [comment]
Looks good, thanks! I assume it was something like `s/^([A-Za-z_][A-Za-z0-9_@#]*): MACRO/MACRO \1/g`?

Please add `DEF` to constants in this PR as well. I thought that would be too disruptive, but after all, most constants are done with `const` and the `DEF`s would mostly be within macros anyway. A regex like `s/^([A-Za-z_][A-Za-z0-9_@#]*\s+(?:EQU\b|EQUS\b|=|rb\b|rw\b|rl\b))/DEF \1/g` should cover all of them, but then it'll be nice to manually indent ones inside of macros (since that's one reason rgbasm introduced the `DEF` syntax anyway).

**Comment by vulcandth** [comment]
@Rangi42 I added DEF in front of the constants. I have not yet indented the constants in the MACRO's. I plan to go back through an manually tackling that. Could you please give one or two examples of how you like it indented.. and i'll apply it to the rest.

**Comment by Rangi42** [comment]
Oops, I forgot to get the new `+=`/`-=`/etc operators in that regex (like `__deco_value__ += 1`).

https://github.com/pret/pokecrystal/commit/ab0f9c287a45d1986985545d7ec6a2cd862e26fc

**Comment by Idain** [comment]
I still think that Regex is advanced sorcery, so I'm very impressed.

**Comment by Rangi42** [comment]
I think this is ready now. @aaaaaa123456789 @mid-kid want to review?

> I still think that Regex is advanced sorcery, so I'm very impressed.

They're literally simpler than complete programming languages, it's just that they have a very dense syntax, and it can be hard to read which characters are regex language operators vs which are literal text. Python has a verbose regex mode that lets you space things out and add comments, like:

```
^                   # Start at the beginning of the line
(                   # Capture all this text so we can refer to it in the 'DEF \1' replacement
  [A-Za-z_]         # Constants start with a letter or '_'
  [A-Za-z0-9_@#] *  # Constants may then have letters, digits, '_'s, '@'s, or '#'s
  \s +              # At least one whitespace character between the constant and the operator
  [-+*/%<>&|^]      # The first character of '-=', '+=', '*=', etc
  =                 # The second character is always an '='
)
```

People have also made various [alternative syntaxes](https://lobste.rs/s/molbhc/alternative_regex_syntax) that some might find easier to read/write.

File: macros/vc.asm [hunk 1]
```diff
@@ -1,26 +1,26 @@
-vc_hook: MACRO
+MACRO vc_hook
```
Comment (aaaaaa123456789) [inline macros/vc.asm:1]: Everything in this file needs reformatting (reindenting, etc.).

File: maps/CeladonGameCornerPrizeRoom.asm [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-CELADONGAMECORNERPRIZEROOM_TM15_COINS EQU 7500
-CELADONGAMECORNERPRIZEROOM_PIKACHU_COINS  EQU 2222
-CELADONGAMECORNERPRIZEROOM_PORYGON_COINS  EQU 5555
-CELADONGAMECORNERPRIZEROOM_LARVITAR_COINS EQU 8888
+DEF CELADONGAMECORNERPRIZEROOM_TM32_COINS EQU 1500
+DEF CELADONGAMECORNERPRIZEROOM_TM29_COINS EQU 3500
+DEF CELADONGAMECORNERPRIZEROOM_TM15_COINS EQU 7500
+DEF CELADONGAMECORNERPRIZEROOM_PIKACHU_COINS  EQU 2222
+DEF CELADONGAMECORNERPRIZEROOM_PORYGON_COINS  EQU 5555
+DEF CELADONGAMECORNERPRIZEROOM_LARVITAR_COINS EQU 8888
```
Comment (aaaaaa123456789) [inline maps/CeladonGameCornerPrizeRoom.asm:6]: These values should all be aligned.

File: maps/GoldenrodGameCorner.asm [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-GOLDENRODGAMECORNER_TM38_COINS EQU 5500
-GOLDENRODGAMECORNER_ABRA_COINS      EQU 100
-GOLDENRODGAMECORNER_CUBONE_COINS    EQU 800
-GOLDENRODGAMECORNER_WOBBUFFET_COINS EQU 1500
+DEF GOLDENRODGAMECORNER_TM25_COINS EQU 5500
+DEF GOLDENRODGAMECORNER_TM14_COINS EQU 5500
+DEF GOLDENRODGAMECORNER_TM38_COINS EQU 5500
+DEF GOLDENRODGAMECORNER_ABRA_COINS      EQU 100
+DEF GOLDENRODGAMECORNER_CUBONE_COINS    EQU 800
+DEF GOLDENRODGAMECORNER_WOBBUFFET_COINS EQU 1500
```
Comment (aaaaaa123456789) [inline maps/GoldenrodGameCorner.asm:6]: These values should all be aligned.

File: mobile/mobile_42.asm [hunk 4]
```diff
@@ -1474,7 +1474,7 @@ Function108b78:
 	ld [hld], a
 	pop af
 	ldh [rSVBK], a
-	ld a, TRUE 
+	ld a, TRUE
```
Comment (aaaaaa123456789) [inline mobile/mobile_42.asm:1477]: Like with the other comment, whitespace pruning should be separate.

File: wram.asm [hunk 5]
```diff
@@ -2233,7 +2233,7 @@ wBattleMenuCursorPosition:: db
 
 	ds 1
 
-wCurBattleMon:: 
+wCurBattleMon::
```
Comment (aaaaaa123456789) [inline wram.asm:2236]: Same as before: keep whitespace pruning separate.

**Review by aaaaaa123456789 — CHANGES_REQUESTED** [review]
Noticed a few minor things. Nice work!

File: engine/movie/intro.asm [hunk 6]
```diff
@@ -1438,19 +1438,19 @@ endr
...
 endr
 
 .SlowFadePalettes:
-hue = 31
+DEF hue = 31
 rept 16
 	RGB hue, hue, hue
-hue -= 1
+DEF hue -= 1
 endr
```
[the remark this answers is not in the document]
Comment (vulcandth) [reply engine/movie/intro.asm:1454]:
```assembly
.FastFadePalettes:
DEF hue = 31
rept 8
	RGB hue, hue, hue
	DEF hue -= 1
	RGB hue, hue, hue
	DEF hue -= 2
endr
```
Like this?

**Comment by Rangi42** [comment]
Updating the `MACRO` and `DEF` syntaxes here looks good to me. It should wait to merge until all four repos are ready.

File: sram.asm [hunk 7]
```diff
@@ -174,10 +174,10 @@ endr
...
 ; The PC boxes will not fit into one SRAM bank,
 ; so they use multiple SECTIONs
-box_n = 0
-boxes: MACRO
+DEF box_n = 0
+MACRO boxes
 rept \1
-box_n += 1
+	DEF box_n += 1
 sBox{d:box_n}:: box sBox{d:box_n}
```
Comment (vulcandth) [inline sram.asm:181]: This line should be indented.

**Review by vulcandth — COMMENTED** [review]
Noticed a couple things that still need fixing while doing pokegold.

**Pull request merged**